contrib/kind: adapt clustermesh related make targets to recent changes

giorio94 opened the pull request:
This PR fixes two issues concerning the clustermesh related make targets introduced by recent modifications:

* `make kind-clustermesh`: do not fail if the `kind-cilium` docker network already exists
* `make kind-install-cilium-clustermesh`: initialize the `LOCAL_CLUSTERMESH_IMAGE` variable

Additionally, it extends the kind-down.sh script and makes the `kind-clustermesh-down` target use it, ensuring that the `kind-cilium` network gets deleted also in that case.

More details are provided in the respective commit messages.

<!-- Description of change -->

```release-note
contrib/kind: adapt clustermesh related make targets to recent changes
```

giorio94 committed:
contrib/kind: do not fail if the kind-cilium network already exists

4310b5e5c4a1 modified the kind.sh contrib script introducing the
creation of a dedicated kind-cilium docker network to be used by kind.
Yet, this broke the `make kind-clustermesh` command, as the second
execution of the script fails given that the network already exists.
Hence, let's first check if the network already exists, and create it
only if not already present.

Fixes: 4310b5e5c4a1 ("contrib/kind: enable XDP_TX from pod veth")
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
contrib/kind: fix uninitialized variable in install-clustermesh target

2975c75613a1 dropped the usage of a local docker registry and, among
others, changed the `kind-install-cilium-clustermesh` target to use a
parametrized name for the clustermesh-apiserver image. Yet, that
variable was not initialized, causing the target to fail. This commit
fixes it, calling the initialization logic.

Fixes: 2975c75613a1 ("contrib/kind: no longer create local docker registry")
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

bimmlerd commented on Makefile:
```diff
@@ -431,6 +431,7 @@ kind-clustermesh: ## Create two kind clusters for clustermesh development.
 kind-clustermesh-down: ## Destroy kind clusters for clustermesh development.
 	kind delete clusters clustermesh1
 	kind delete clusters clustermesh2
+	docker network rm kind-cilium
```
could we change kind-down.sh to take clusters to delete as optional argument to not duplicate the logic and for symmetry? Then this could become just `kind-down.sh clustermesh1 clustermesh2`. WDYT?

giorio94 replied:
Done

bimmlerd reviewed (changes requested):
I'd prefer changing kind-down instead of duplicating it's logic in the Makefile

giorio94 committed:
contrib/kind: cleanup kind-cilium network on kind-clustermesh-down

This commit parametrizes the `kind-down.sh` script, to tear down all
kind clusters specified as parameters (defaulting to kind if none are
specified), and changes the `kind-clustermesh-down` target to use it.
Hence, uniforming the cleanup logic to reduce possible discrepancies.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

marseel reviewed (approved):
Looks great, thanks!

squeed reviewed (approved):
Sorry for the churn, thanks for cleaning up my mess :p

lmb commented on Makefile:
```diff
@@ -459,7 +458,7 @@ kind-clustermesh-images: kind-clustermesh-ready kind-build-clustermesh-apiserver
 	$(QUIET)kind load docker-image $(LOCAL_OPERATOR_IMAGE) --name clustermesh1
 	$(QUIET)kind load docker-image $(LOCAL_OPERATOR_IMAGE) --name clustermesh2
 
-.PHONY: kind-install-cilium-clustermesh
```
So `call KIND_ENV` ends up doing the same as `.PHONY`?

giorio94 replied:
Yes, it adds `.PHONY` among the others: https://github.com/cilium/cilium/blob/c3cb8eba315b8786a166e0c34bae78f7fd202bf4/Makefile#L444-L450

bimmlerd reviewed (approved):
thanks!
